### PR TITLE
Embind: Simplify Integer toWireType function

### PR DESCRIPTION
**Pull request by billbudge**
Checks integer type and range assertions only for ASSERTIONS builds.

Checks the integer isUnsignedType at function creation time to speed up runtime code. Removes the value | 0 bit, as the VM will convert ToInt32 for signed integers. This assumes that this registration code is only invoked if the Wasm function argument type is i32. Otherwise, we should restore the "| 0" to get the same behavior.

Leave BigInt unchanged for now.

Cleans up some whitespace issues in the neighborhood.

### CODE REVIEW

**Comment by billbudge** [comment]
I will have to look at the failing tests.

File: src/embind/embind.js [hunk 1]
```diff
@@ -572,21 +572,34 @@ var LibraryEmbind = {
...
+        if (value < minRange || value > maxRange) {
+            throw new TypeError('Passing a number "' + _embind_repr(value) + '" from JS side to C/C++ side to an argument of type "' + name + '", which is outside the valid range [' + minRange + ', ' + maxRange + ']!');
+        }
+#endif
+    }
+    var toWireType;
+    if (isUnsignedType) {
+        toWireType = function(destructors, value) {
+            checkAssertions(value, this.name);
+            return (value >>> 0);
```
Comment (sbc100) [inline src/embind/embind.js:589]: No need for the `( )` anymore I guess

Comment (billbudge) [reply src/embind/embind.js]: Yes, fixed it.

File: tests/embind/embind.test.js [hunk 2]
```diff
@@ -638,11 +638,13 @@ module({
...
-        assert.throws(TypeError, function() { cm.const_ref_adder(Symbol('0'), 1); });
-        assert.throws(TypeError, function() { cm.const_ref_adder(BigInt(0), 1); });
+        test("passing Symbol or BigInt as floats always throws", function() {
+            assert.throws(TypeError, function() { cm.const_ref_adder(Symbol('0'), 1); });
+            assert.throws(TypeError, function() { cm.const_ref_adder(BigInt(0), 1); });
+        });
 
-        if (cm['ASSERTIONS']) {
-            test("can pass only number and boolean with assertions assertions", function() {
+        if (cm.ASSERTIONS) {
```
Comment (kripken) [inline tests/embind/embind.test.js:646]: Why is this changed? (it breaks the consistency of how this is used here; and, in general for closure it is right to use the quotes)

Comment (billbudge) [reply tests/embind/embind.test.js:646]: I went back and forth, this should not have been changed for consistency as you state.

Comment (sbc100) [reply tests/embind/embind.test.js:646]: IIUC this code is running outside of the emscripten-generated code and outside of the scope of closure so (I think).. the quotes are maybe not needed?   But yes we should be consistent.. maybe a separate PR to remove all the quotes like this?

Comment (kripken) [reply tests/embind/embind.test.js:646]: If we want the test to run under closure, then it should be quoted. And I think that's a good idea, to test embind in that mode. So I'd go with quotes.

Comment (sbc100) [reply tests/embind/embind.test.js:646]: I must be misunderstanding.  I thought closure only ran on code that was compiled with the module (i.e. --js-library code or EM_ASM code).   This code IIUC is completely external an separate and added/injected *after* compilation, so there is no way it can be effected by closure.. or am I mistaken?

Comment (kripken) [reply tests/embind/embind.test.js]: Oh, I wasn't aware of how this is used. I assumed if it's in `src` then it is compiled with the rest of the code. If it's an exception like `worker.js` then we don't run closure on it, correct.

Still, it can't hurt to have the quotes, if this is just test and not production code.

Comment (sbc100) [reply tests/embind/embind.test.js]: My understanding that this is test code that hosts the emscripten module...  i.e. it lives on the outside and calls into the compiled module.  So there is no closure involved.

**Comment by billbudge** [comment]
I've updated the title and description to reflect the reduced scope, as BigInt is hard to test.

File: src/embind/embind.js [hunk 3]
```diff
@@ -55,12 +55,12 @@ var LibraryEmbind = {
...
 #endif
-#if ASSERTIONS
-    Module['ASSERTIONS'] = true;
-#endif
 #if EMBIND_STD_STRING_IS_UTF8
     Module['EMBIND_STD_STRING_IS_UTF8'] = true;
 #endif
+#if ASSERTIONS
+    Module['ASSERTIONS'] = true;
+#endif
```
Comment (kripken) [inline src/embind/embind.js:63]: This code movement seems to have no effect - do we need it? (Is there some other benefit like aesthetics? Sorry if I'm missing that.)

**Review by kripken — APPROVED** [review]
lgtm % comment

File: src/embind/embind.js (hunk 3 above)
Comment (billbudge) [reply src/embind/embind.js:63]: It's a stray change while I was trying to figure out some test problems. I can move it back, or would it be better to make it first, so the properties are alphabetically sorted?

Comment (kripken) [reply src/embind/embind.js]: I don't have strong feelings - either way sgtm. Mainly I wanted to check if I was missing something.

Comment (billbudge) [reply src/embind/embind.js]: I made it alphabetical order, though I hope we never have enough such properties that it actually matters. Thanks for reviewing!
